django/db/models/fields/related.py

```diff
@@ -731,26 +732,23 @@ def get_reverse_path_info(self):
         pathinfos = [PathInfo(from_opts, opts, (opts.pk,), self.remote_field, not self.unique, False)]
         return pathinfos
 
-    def get_lookup(self, lookup_name):
-        if lookup_name == 'in':
-            return RelatedIn
-        elif lookup_name == 'exact':
-            return RelatedExact
-        elif lookup_name == 'gt':
-            return RelatedGreaterThan
-        elif lookup_name == 'gte':
-            return RelatedGreaterThanOrEqual
-        elif lookup_name == 'lt':
-            return RelatedLessThan
-        elif lookup_name == 'lte':
-            return RelatedLessThanOrEqual
-        elif lookup_name == 'isnull':
-            return RelatedIsNull
-        else:
-            raise TypeError('Related Field got invalid lookup: %s' % lookup_name)
-
-    def get_transform(self, *args, **kwargs):
-        raise NotImplementedError('Relational fields do not support transforms.')
+    def _get_lookup(self, lookup_name):
+        try:
+            return self.class_lookups[lookup_name]
+        except KeyError:
+            # To allow for inheritance, check parent class' class_lookups.
+            for parent in inspect.getmro(self.__class__):
+                # Related lookups need to be explicitly registered to related fields.
+                if parent is ForeignObject:
+                    break
+                if 'class_lookups' not in parent.__dict__:
```

Comment:
Any recommendations or examples to work from? I ran a simple timeit test, and it looks like `get_lookups` performs worse (which makes sense). 


```
s = "models.DataPoint._meta.get_field('name').get_lookup('in')"
t = "models.RelatedPoint._meta.get_field('data').get_lookup('in')"

get_lookups:
s result [0.7386798858642578, 0.736055850982666, 0.7420699596405029, 0.7442851066589355, 0.7473750114440918]
t result [0.6083571910858154, 0.5898189544677734, 0.6105849742889404, 0.6074070930480957, 0.5925500392913818]

bases:
s result [0.2692680358886719, 0.2704458236694336, 0.2689228057861328, 0.2720210552215576, 0.2666950225830078]
t result [0.40657591819763184, 0.3987290859222412, 0.4173128604888916, 0.39420008659362793, 0.3984808921813965]
```